checks/process.go

```diff
@@ -308,6 +316,20 @@ func skipProcess(
 	return false
 }
 
+// ctrForPid uses lastCidForPid and filter down only the pid -> cid that we need
+func (p *ProcessCheck) ctrForPid(pids []int32) map[int32]string {
```

Comment:
This duplication also leaves a bad smell. Why can't we just use the above extracted function?

Reply:
They are different in nature. Previous function takes containers and compose relationship, this function is filtering to only relevant entries in the result. I'm going to change this function name to be something like `filterContainersByPid`.